public/assets/scripts/main.js

```diff
@@ -1,3 +1,127 @@
 //
 // Main JavaScript file
 // --------------------------------------------------------
+var PLTFRM = {
+  init: function () {
+    (window.innerWidth < 768) ? PLTFRM.Sliders.init() : PLTFRM.Clicker.init();
+    this.Resizer.init();
+    this.Retina.init();
+  },
+
+  Clicker: {
+    init: function () {
+      document.querySelector(this.data.selector).classList.add('active');
+      this.data.swipe = Swipe(PLTFRM.Phones.clickEl, PLTFRM.Sliders.options);
+      this.addListeners();
+    },
+    data: {
+      selector: '.functionality',
+      swipe: null
+    },
+    addListeners: function () {
+      var triggers = document.querySelectorAll(this.data.selector);
+      var triggers_l= triggers.length;
+      for(var i = 0; i < triggers_l; i++) {
+        triggers[i].setAttribute('data-index', i);
+        triggers[i].addEventListener('click', this.handleClick, false);
+      }
+    },
+    handleClick: function () {
+      PLTFRM.Clicker.removeActive(index);
+      var index = this.getAttribute('data-index');
+      PLTFRM.Clicker.goToPhone(index);
+      this.classList.add('active');
+    },
+    removeActive: function() {
+      document.querySelector(this.data.selector+'.active').classList.remove('active');
+    },
+    goToPhone: function(index) {
+      this.data.swipe.slide(index, 500);
+    }
+
+  },
+
+  Phones: {
+    swipeEl: document.querySelector("#slider"),
+    clickEl: document.querySelector('#bigPhonePic')
+  },
+
+  Sliders: {
+    init: function () {
+       this.data.swipe = Swipe(PLTFRM.Phones.swipeEl, PLTFRM.Sliders.options);
+    },
+    data: {
+      swipe: null
+    },
+    options: {
+      startSlide: 0,
+      speed: 400,
+      draggable: false,
+      continuous: false,
+      disableScroll: false,
+      stopPropagation: true,
+      callback: function(index, elem, dir) {},
+      transitionEnd: function(index, elem) {}
+
+    }
+  },
+
+  Resizer: {
+    init: function () {
+      window.onresize = function () {
+        if(window.innerWidth < 768 && PLTFRM.Sliders.data.swipe === null) {
+          PLTFRM.Clicker.data.swipe.kill();
+          PLTFRM.Clicker.data.swipe = null;
+          PLTFRM.Sliders.init();
+        } else if(window.innerWidth >= 768 && PLTFRM.Clicker.data.swipe === null) {
+          setTimeout(function() {
+            PLTFRM.Sliders.data.swipe.kill();
+            PLTFRM.Sliders.data.swipe = null;
+          }, 200);
+          PLTFRM.Clicker.init();
+        }
+      }
+    }
+  },
+
+  Retina: {
+    init: function() {
+      if (window.devicePixelRatio > 1) {
+        var images  = document.querySelectorAll('img');
+        var images_l = images.length;
+        for (var i = 0; i < images_l; i++) {
+          var splitedImages = images[i].src.split(".");
+          if(splitedImages[1] == "jpg" || splitedImages[1] == "png") {
+            images[i].src = splitedImages[0]+"@2x."+splitedImages[1];
```

Comment:
So I like this solution because it gives us support for IE and Edge, however, the problem is that we now actually make 2 network requests for each image. The way you can see this is if you open up the Chrome console, and open the Network tab, you'll notice a request for both each normal image and @2x image. 

I'd say a good solution for this is to use srcset on all img tags, and then use feature detection here to only swap the src in the JS if the browser doesn't support srcset. The way to do this is use a feature detection library. However, since this is a really specific and simple case, we can also just go ahead and detect IE11 or Edge browsers, and not have to bother with a feature detection library. 

Here is a IE detection code snippet you can add to main.js above `PLTFRM.init();`:
```
var Utils = {
  detectIE: function() {
    var ua = window.navigator.userAgent;

    var trident = ua.indexOf('Trident/');
    if (trident > 0) {
      // IE 11 => return version number
      var rv = ua.indexOf('rv:');
      return parseInt(ua.substring(rv + 3, ua.indexOf('.', rv)), 10);
    }

    var edge = ua.indexOf('Edge/');
    if (edge > 0) {
      // Edge (IE 12+) => return version number
      return parseInt(ua.substring(edge + 5, ua.indexOf('.', edge)), 10);
    }

    return false;
  }
};
```

And then do determine if you should img swap, you can use it like this:
```
if (Utils.detectIE()) {
    images[i].src = splitedImages[0]+"@2x."+splitedImages[1];
}
```